### PR TITLE
docs: update Swift docs for v1 release

### LINKED ISSUE #109: Add documentation for RPC returning data
# Improve documentation

## Link 

[Add a link to the page which needs improvement (if relevant)](https://supabase.com/docs/reference/swift/using-filters)

## Describe the problem

The documentation could use an example showing how to use data returned from a Postgres function. I have a function returning data and it was non trivial to figure out how to get that data.

## Describe the improvement

I'd just add another code example named "Calling a function that returns data". Here's an example line from my project:
```
let emailExists:Bool = try await sbClient.database.rpc(fn: "check_if_email_exists", params: ["email": email]).execute().value
```

## Additional context

n/a

### LINKED ISSUE #134: Add Documentation for PostgREST

### LINKED ISSUE #135: Add Documentation for Realtime

### LINKED ISSUE #136: Add Documentation for Storage

### LINKED ISSUE #137: Add Documentation for Functions

### PR DESCRIPTION
**Pull request by grdsdev**
## What kind of change does this PR introduce?

Docs update

Close https://github.com/supabase-community/supabase-swift/issues/109
Close https://github.com/supabase-community/supabase-swift/issues/134
Close https://github.com/supabase-community/supabase-swift/issues/135
Close https://github.com/supabase-community/supabase-swift/issues/136
Close https://github.com/supabase-community/supabase-swift/issues/137

## What is the current behavior?

Currently have outdated documentation for Swift library

## What is the new behavior?

Update Swift documentation with the latest changes from the upcoming v1 release.

### CODE REVIEW

File: spec/supabase_swift_v1.yml [hunk 1]
```diff
@@ -0,0 +1,3463 @@
...
+            phone: "+13334445555",
+            token: "123456",
+            type: .sms
+          )
+          ```
+  - id: sign-in-with-otp
+    title: "signInWithOTP()"
+    notes: |
+      - This method is used for passwordless sign-ins where a OTP is sent to the user's email or phone number.
+      - If the user doesn't exist, `signInWithOTP()` will signup the user instead. To restrict this behaviour, you can set `shouldCreateUser` to `false`.
```
Comment (dshukertjr) [inline spec/supabase_swift_v1.yml:170]:
```suggestion
      - If the user doesn't exist, `signInWithOTP()` will signup the user instead. To restrict this behavior, you can set `shouldCreateUser` to `false`.
```

File: spec/supabase_swift_v1.yml [hunk 2]
```diff
@@ -0,0 +1,3463 @@
...
+          By default, the data is returned in JSON format, but can also be returned as Comma Separated Values.
+        hideCodeBlock: true
+        isSpotlight: true
+  - id: invoke
+    title: invoke()
+    description: |
+      Invoke a Supabase Edge Function.
+    notes: |
+      - Requires an Authorization header.
+      - When you pass in a body to your function, we automatically attach the Content-Type header for `String`, and `Data`. If it doesn't match any of these types we assume the payload is `json`, serialise it and attach the `Content-Type` header as `application/json`. You can override this behaviour by passing in a `Content-Type` header of your own.
```
Comment (dshukertjr) [inline spec/supabase_swift_v1.yml:2660]:
```suggestion
      - When you pass in a body to your function, we automatically attach the Content-Type header for `String`, and `Data`. If it doesn't match any of these types we assume the payload is `json`, serialise it and attach the `Content-Type` header as `application/json`. You can override this behavior by passing in a `Content-Type` header of your own.
```

Comment (dshukertjr) [inline spec/supabase_swift_v1.yml:2660]:
```suggestion
      - When you pass in a body to your function, we automatically attach the Content-Type header for `String`, and `Data`. If it doesn't match any of these types we assume the payload is `json`, serialize it and attach the `Content-Type` header as `application/json`. You can override this behaviour by passing in a `Content-Type` header of your own.
```

**Review by dshukertjr — APPROVED** [review]

**Comment by dshukertjr** [comment]
Thanks for this! Looks great! Feel free to merge whenever you want to!

**Review by kangmingtay — APPROVED** [review]

**Review by J0 — APPROVED** [review]

**Comment by samuelOsborne** [comment]
hey @dshukertjr Is it here: https://supabase.com/docs/reference/swift/ that we can see these changes?

**Comment by grdsdev** [comment]
That's correct @samuelOsborne

**Comment by dshukertjr** [comment]
I think the confusion might be coming from the docs label saying v0.0. 

@grdsdev I think this is the last missing piece for the docs update: https://github.com/supabase/supabase/pull/18931

**Pull request merged**